src/ruby/pb/test/xds_client.rb

```diff
@@ -282,12 +258,15 @@ def run_test_loop(stub, target_seconds_between_rpcs, fail_on_failed_rpcs)
       else
         raise "Unsupported rpc #{rpc}"
       end
-      rpc_stats_key = rpc.to_s
-      if metadata.key?('rpc-behavior') or metadata.key?('fi_testcase')
-        keep_open_threads << execute_rpc_in_thread(op, rpc_stats_key)
-      else
-        results[rpc] = execute_rpc(op, fail_on_failed_rpcs, rpc_stats_key)
+      keep_open_threads << execute_rpc_in_thread(op, rpc)
```

Comment:
It looks pre-existing so doesn't need to block this PR, but one potential thing I'm a little weary of here is that if QPS is set high enough to saturate the client's QPS capability, then we'll be opening threads faster than the rate at which we're completing them, which could lead to OOM.